### PR TITLE
feat(next): introduce the openapi generated client in the next package

### LINKED ISSUE #150: Prepare OpenAPI-generated go-katapult client in `next` branch
Now that Katapult exposes a [experimental OpenAPI schema](https://api.katapult.io/core/v1/schema/openapi.json), we should start looking at generating a new Go client for Katapult's API from the OpenAPI schema.

Initially I believe it would best to keep this new client in a branch called `next`, as it will be complete departure from the current codebase.

One thing to keep in mind, is the `buildspec` package, as the OpenAPI schema simply uses a String type for the XML spec, the buildspec package would still be very useful for programmatically building VM build specs.

### PR DESCRIPTION
**Pull request by ikadix**
Now available in `github.com/krystal/go-katapult/next` is a complete client for the katapult API.
 This is generated from the openapi spec. 
For now this includes a docker image and a shell script for the build command. 

> [!NOTE]  
> This bumps the following things :
> - Go Version is now minimum `v1.20`
> - Golangci-lint is now using `v1.56`
> - checkout action is now `v4` 
> - setup-go action is now `v5`
> - golangci-lint action is now in use with `v4`
> - codeclimate action is now `v5`
>
> `goconst` has been disabled in test files. 
> a few changes have been made across the project to remove usage of `io/ioutil`

Closes #150

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,71 +7,45 @@ jobs:
...
-          go-version: "1.18"
-      - uses: actions/cache@v3
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          go-version: "1.20"
       - name: Publish coverage
         uses: paambaati/codeclimate-action@v3.2.0
```
Comment (jimehk) [inline .github/workflows/ci.yml:50]: We should bump this to v5.0.0 as well:
```suggestion
        uses: paambaati/codeclimate-action@v5.0.0
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
```
Comment (jimehk) [inline .github/workflows/nightly.yml:14]: These should also be updated:

```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-go@v5
```

File: README.md [hunk 3]
```diff
@@ -46,3 +46,34 @@ Documentation:
 
 - [API Documentation](https://developers.katapult.io/api/docs/latest/)
 - [Katapult Documentation](https://docs.katapult.io/)
+
+
+
+# Next Client
```
Comment (jimehk) [inline README.md:52]: Nitpick:

```suggestion
# Experimental "next" Client 
```

File: next/katapult-openapi.json
Comment (jimehk) [inline next/katapult-openapi.json:1]: Do we have any script and/or Make target that can update this file from the production URL?

**Review by jimehk — CHANGES_REQUESTED** [review]
Couple of minor tweaks to the CI workflows are needed, and I have a question about the bundled OpenAPI schema. Overall looks good and promising though :)

File: next/katapult-openapi.json
Comment (ikadix) [reply next/katapult-openapi.json:1]: Added make targets for getting the schema from the url and for running the generate script

**Review by jimehk — APPROVED** [review]

**Pull request merged**